ui/src/app/shared/components/revision.tsx

```diff
@@ -1,9 +1,12 @@
 import * as React from 'react';
 import {revisionUrl} from './urls';
 
-export const Revision = ({repoUrl, revision, children}: {repoUrl: string; revision: string; children?: React.ReactNode}) => {
+export const Revision = ({repoUrl, revision, path, children}: {repoUrl: string; revision: string; path?: string; children?: React.ReactNode}) => {
     revision = revision || '';
-    const url = revisionUrl(repoUrl, revision);
+    let url = revisionUrl(repoUrl, revision);
+    if (path) {
+        url += '/' + path;
```

Comment:
I think the main trick will be making sure this works/makes sense in the major SCM providers (GitHub, Gitlab, etc.). I'm excited that the feature looks fairly simple to implement!

Reply:
NP, I check that on GitHub, let me check that this is work on GitLab + bitbucket